libraries/standard/mqtt/src/iot_mqtt_api.c

```diff
@@ -280,6 +280,35 @@ static void _handleConnectFailure( IotMqttConnection_t pMqttConnection,
                                    _mqttOperation_t * pOperation,
                                    bool ownNetworkConnection );
 
+/**
+ * @brief Send a Publish asynchronously by scheduling operations on the taskpool.
+ * @param[in] mqttConnection The MQTT connection to use for the PUBLISH.
+ * @param[in] pPublishInfo MQTT publish parameters.
+ * @param[in] flags Flags which modify the behavior of this function.
+ * @param[in] pCallbackInfo Asynchronous notification of this function's completion (`NULL` to disable).
+ * @param[out] pPublishOperation Set to a handle by which this operation may be
+ * referenced after this function returns. This reference is invalidated once
+ * the publish operation completes.
+ *
+ * @return This function will return #IOT_MQTT_STATUS_PENDING upon success for
+ * QoS 1 publishes
```

Comment:
How about the return codes when there are errors?

Reply:
Added documentation.